src/moscot/base/cost.py

```diff
@@ -0,0 +1,70 @@
+from abc import ABC, abstractmethod
+from typing import Any, Tuple, Union
+
+import numpy as np
+
+from anndata import AnnData
+
+from moscot._logging import logger
+from moscot._types import ArrayLike
+
+__all__ = ["BaseCost"]
+
+
+class BaseCost(ABC):
+    """Base class for all :mod:`moscot` losses.
+
+    Parameters
+    ----------
+    adata
+        Annotated data object.
+    attr
+        Attribute of :class:`anndata.AnnData` used when computing the cost.
+    key
+        Key in the ``attr`` of :class:`anndata.AnnData` used when computing the cost.
+    dist_key
+        Helper key which determines into which subset ``adata`` belongs.
```

Comment:
maybe 
`Helper key which determines which distribution ``adata`` belongs to.` ? 
or
`Identifier defining the key of the distribution ``adata`` belongs to.`?

Reply:
Thanks, used the former!